### PR TITLE
Block editor: fix focus handler in Safari

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Handlers on a non-selected blocks could be attached with a delay, so always attach the handler.

Different from #30968 because we leave the other handlers to only attach on selected blocks. Let's see if there's any test failures in this case.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/e2e-tests/specs/editor/blocks/list.test.js [hunk 1]
```diff
@@ -113,6 +113,7 @@ describe( 'List', () => {
 		await clickBlockAppender();
 		await page.keyboard.press( 'Enter' );
 		await page.keyboard.type( '* ' );
+		await page.evaluate( () => new Promise( window.requestIdleCallback ) );
```
Comment (ellatrix) [inline packages/e2e-tests/specs/editor/blocks/list.test.js:116]: @youknowriad I found the issue I think. Feel free to include in your PR. I don't mind which PR we go with. We need to wait for this:

https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/block-editor/src/store/controls.js#L24-L36

**Review by youknowriad — APPROVED** [review]
This looks good to me. 

I think conditionally attaching event handlers for blocks though is probably something we should consider for all event  handlers on blocks because of  potential hidden async mode issues. That said this PR does solve the important one that we're aware of.

**Comment by ellatrix** [comment]
Thanks for the review!

> I think conditionally attaching event handlers for blocks though is probably something we should consider for all event handlers on blocks because of potential hidden async mode issues.

Right, if the blocks is not selected, but the remaining ones are for the selected block, so that will attach instantly.

**Comment by ellatrix** [comment]
The performance is apparently improved... 🤔 I think that's just because at a different time the tests run faster or slower.

```
┌──────────────────┬──────────────────────────────────────────┬──────────────┐
298
│     (index)      │ 89a85fc4f19ee176ee9201e3f1a80f0c40a51122 │    trunk     │
299
├──────────────────┼──────────────────────────────────────────┼──────────────┤
300
│       load       │               '10610.4 ms'               │ '11140.2 ms' │
301
│       type       │                '50.58 ms'                │  '57.35 ms'  │
302
│     minType      │                '42.79 ms'                │  '46.25 ms'  │
303
│     maxType      │               '369.79 ms'                │ '421.56 ms'  │
304
│      focus       │               '130.88 ms'                │ '135.02 ms'  │
305
│     minFocus     │                '97.22 ms'                │  '98.49 ms'  │
306
│     maxFocus     │               '293.71 ms'                │ '297.48 ms'  │
307
│   inserterOpen   │               '897.59 ms'                │ '921.11 ms'  │
308
│ minInserterOpen  │               '809.31 ms'                │ '851.37 ms'  │
309
│ maxInserterOpen  │               '1303.15 ms'               │ '1367.68 ms' │
310
│  inserterHover   │                '39.53 ms'                │  '44.45 ms'  │
311
│ minInserterHover │                '35.07 ms'                │  '36.46 ms'  │
312
│ maxInserterHover │                '50.42 ms'                │  '53.58 ms'  │
313
└──────────────────┴──────────────────────────────────────────┴──────────────┘
```

**Pull request merged**